crates/ra_hir_expand/src/db.rs

```diff
@@ -97,8 +132,40 @@ pub(crate) fn parse_macro(
             log::warn!("fail on macro_parse: (reason: {})", err,);
         })
         .ok()?;
-    match macro_file.macro_file_kind {
-        MacroFileKind::Items => mbe::token_tree_to_items(&tt).ok().map(Parse::to_syntax),
-        MacroFileKind::Expr => mbe::token_tree_to_expr(&tt).ok().map(Parse::to_syntax),
-    }
+    let res = match macro_file.macro_file_kind {
+        MacroFileKind::Items => {
+            mbe::token_tree_to_items(&tt.0).ok().map(|(p, map)| (Parse::to_syntax(p), map))
+        }
+        MacroFileKind::Expr => {
+            mbe::token_tree_to_expr(&tt.0).ok().map(|(p, map)| (Parse::to_syntax(p), map))
+        }
+    };
```

Comment:
Here and elsewhere, it seems like code can be simplified by using `?` and destructing

```
    let (parsed, exp_map) = match macro_file.macro_file_kind {
        MacroFileKind::Items => {
            mbe::token_tree_to_items(&tt.0).ok().map(|(p, map)| (Parse::to_syntax(p), map))? // ? can be pushed even further inside
        }
        MacroFileKind::Expr => {
            mbe::token_tree_to_expr(&tt.0).ok().map(|(p, map)| (Parse::to_syntax(p), map))?
        }
    };
```

Reply:
Yeah, fixed, thanks